modules/thermal_hydraulics/test/tests/materials/ad_wall_heat_transfer_coefficient_kazimi/kazimi_test.i

```diff
@@ -0,0 +1,90 @@
+#liquid sodium properties at @773 [K]
+rho = 762.90
+vel = 0.1
+k = 64.217
+mu = 2.358e-4
+cp = 1264.6
+T = 773
+T_wall = 774
+D_h = 0.1
+PoD = 1.1
+
+[Mesh]
+  type = GeneratedMesh
+  dim = 1
+  nx = 1
+  allow_renumbering = false
+[]
+
+[Variables]
+  [u]
+  []
+[]
+
+[Kernels]
+  [diff]
+    type = ADDiffusion
+    variable = u
+  []
+[]
+
+[BCs]
+  [left]
+    type = DirichletBC
+    variable = u
+    boundary = left
+    value = 0
+  []
+  [right]
+    type = DirichletBC
+    variable = u
+    boundary = right
+    value = 1
+  []
+[]
+
+[AuxVariables]
+  [Hw]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+[]
+
+[AuxKernels]
+  [Hw_ak]
+    type = ADMaterialRealAux
+    variable = Hw
+    property = Hw
+  []
+[]
+
+[Materials]
+  [props]
+    type = ADGenericConstantMaterial
+    prop_names = 'rho  vel  k  mu  cp  T  T_wall  D_h'
+    prop_values = '${rho}  ${vel}  ${k}  ${mu}  ${cp}  ${T}  ${T_wall}  ${D_h}'
+  []
+
+  [Hw_material]
+    type = ADWallHeatTransferCoefficientKazimiMaterial
+    PoD = ${PoD}
+  []
+[]
+
+[Executioner]
+  type = Steady
+[]
+
+[Postprocessors]
+  [Hw]
+    type = ElementalVariableValue
+    elementid = 0
+    variable = Hw
+  []
+[]
+
+[Outputs]
+  csv = true
+  execute_on = timestep_end
```

Comment:
```suggestion
  execute_on = 'INITIAL'
```

Reply:
done

Reply:
Josh, for this case here if I use "execute_on = 'INITIAL'" the tests fail with CSVDIFF error. With the INITIAL option MOOSE is writing the CSV like [[0,0],[1,0]]. If I use "execute_on = 'FINAL'" the tests pass without problems

Reply:
> With the INITIAL option MOOSE is writing the CSV like [[0,0],[1,0]]

So do you mean there are two rows of numbers in your output file? I'd expect that there would be a header row and then one row of numbers, with a 0 in the time column. And that's why I'd expect a diff, because of the time value you had there before (1), corresponding to your arbitrary time step size.

Reply:
Oh, sorry, I only have one line in the CSV, the first one is the header, but, if I run with INITIAL it returns me like [0,0] instead of [0,value]

Reply:
Fixed! After looking carefully the example you posted here I find the solution, we need to add:

 ```
[GlobalParams]
  execute_on = 'initial'
[]
```

in the input file, so, all objects are solved in the INITIAL time step  and the outputs are returned correctly, I think probably my materials wasn't being calculated in the INITIAL time step, that's is why it was returning 0.

Reply:
Materials don't have an `execute_on` - you can think of them as being executed when they're requested by other objects. The objects that need `execute_on = INITIAL` are your aux kernels and PPs.

Reply:
oh, this makes much more sense! By the way, the Docs: moose_test failed, I take a look in the error message and I don't think I understand what causes the failure and how to fix it